python/taichi/lang/ops.py

```diff
@@ -419,6 +419,9 @@ def bit_shl(a, b):
 def bit_sar(a, b):
     return _binary_operation(ti_core.expr_bit_sar, ops.rshift, a, b)
 
+@binary
+def bit_shr(a, b):
+    return _binary_operation(ti_core.expr_bit_shr, ops.rshift, a, b)
```

Comment:
There's no proper way to implement logical right shift in python scope for **ALL** possible integer widths since python has unlimited integer widths, so probably we should disable this?

Reply:
1. Let's decorate this class with `@taichi_scope` so that it will fail when called from Python scope.
2. Let's do not distinguish SHR and SAR by operator, but by operand types: SHR if type is ti.i32, SAR if type is ti.u32.

Reply:
By operand type is not a good idea IMO, bitwise operators are low-level operations that regardless of the high-level interpretation of bit values.